diff --git a/src/corelib/tools/qhash.h b/src/corelib/tools/qhash.h
index a2d9434..52a5b13 100644
--- a/src/corelib/tools/qhash.h
+++ b/src/corelib/tools/qhash.h
@@ -48,6 +48,27 @@
 #include <QtCore/qlist.h>
 #include <QtCore/qpair.h>
 
+#ifdef ARTEMIS
+namespace artemis {
+
+  class FormInput;
+  class EventDescriptor;
+  class AjaxRequest;
+  class EventHandlerDescriptor;
+  class FormField;
+  class SourceInfo;
+  class ExecutableConfiguration;
+}
+
+uint qHash(const artemis::FormInput &key);
+uint qHash(const artemis::EventDescriptor &d);
+uint qHash(const artemis::AjaxRequest &key);
+uint qHash(const artemis::EventHandlerDescriptor &key);
+uint qHash(const artemis::FormField &key);
+uint qHash(const artemis::ExecutableConfiguration &key);
+#endif
+
+
 QT_BEGIN_HEADER
 
 QT_BEGIN_NAMESPACE
